crates/common_utils/src/types.rs

```diff
@@ -1144,6 +1147,70 @@ where
     }
 }
 
+/// Domain type for NameType, used for Business Profile Name
+#[derive(
+    Debug, Clone, PartialEq, Eq, serde::Deserialize, serde::Serialize, AsExpression, ToSchema,
+)]
+#[diesel(sql_type = sql_types::Text)]
+pub struct NameType(LengthString<MAX_ALLOWED_PROFILE_NAME_LENGTH, 1>);
+
+impl NameType {
+    /// Create a new NameType from a String
+    pub fn from_string(input_str: String) -> Result<Self, error_stack::Report<ValidationError>> {
+        match LengthString::from(input_str.into()) {
+            Ok(lengthstring) => Ok(Self(lengthstring)),
+            Err(_) => Err(report!(ValidationError::InvalidValue {
```

Comment:
change context here, we want to preserve the context from the root error till leaf error